code-challenges.js

```diff
@@ -1,51 +1,148 @@
+// (3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
-
-
-
-
-
-// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
+//psuedocode:   Create an empty array to hold min/max number. 
+//              Create a function to sort array from least to greatest
+//              Push first and last index as min/max values into empty array 
+const minMax = (array) =>{
+    newArr=[]
+    array.sort(function(a, b){return a-b})
+    //nums1=[-8, 0, 3, 6, 23, 56, 90]
+    //nums2=[-59, 5, 8, 9, 24, 109,]
+    newArr.push(array[0])
+    newArr.push(array[array.length-1])
+    return newArr
+}
+// console.log(minMax(nums1));
+// console.log(minMax(nums2));
+// ------------------- Using min/max built-in method ---------------------
+const arrMinMax = (array) => {
+    newArr1=[]
+    newArr1.push(Math.min(...array)) //note: used spread operator
+    newArr1.push(Math.max(...array))
+    return newArr1
+}
+// console.log(arrMinMax(nums1));
+// console.log(arrMinMax(nums2));
+
+//------------------------------------------------------------------------
+// (4) Create a function that takes in a string and returns a string with every other letter capitalized.
 
 var testString1 = "albatross"
 // Expected output: "aLbAtRoSs"
 var testString2 = "jabberwocky"
 // Expected output: "jAbBeRwOcKy"
-
-
-
-
-
-// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
+//pseudocode: Create a function
+// 1. turn string into an array
+// 2. Iterate through array  
+// 3. Use conditional to draw out odd vs even index 
+// 4. Use uppercase/lowercase method
+// 5. join array into string
+
+const caps = (string) => {
+    let strArr= string.toLowerCase().split("")
+		newArr= strArr.map ((value, index) => {
+			if (index%2 !==0) {
+                return value.toUpperCase()
+            } else {
+                return value
+            } 
+})
+        return newArr.join("")
+}
+console.log(caps(testString1));
+console.log(caps(testString2));
+
+//------------------------------------------------------------------------
+// (5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
 
 var arr1 = [3, 7, 10, 5, 4, 3, 3]
 var arr2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+var arr1 = [3, 7, 10, 5, 4, 3]
+var arr2 = [7, 8, 2, 1, 5, 4]
+deDup1 = (arr1, arr2) => {
+  let combo = arr1.concat(arr2)
+  let noDups = combo.filter((value, index) => {
+    return combo.indexOf(value) === index
+  })
+  return noDups
+}
+console.log(deDup1(arr1, arr2))
```

Comment:
This is great! 

Something I want to encourage you to do is consider your use of variables. As we build larger and more complex projects, we want to start being conscious of precious memory usage and runtime associated with creating unnecessary variables/returns/code.

I refactored your code just SLIGHTLY, and I'd like you to take a look at how we can achieve the same result with just a few changes.

`deDup1 = (arr1, arr2) => {
  let combo = arr1.concat(arr2)
  return combo.filter((value, index) => { return combo.indexOf(value) === index })
}`

Again, your solution is awesome. Just something I want you to think about for the future, especially when you're working with big projects!